packages/ra-ui-materialui/src/input/AutocompleteInput.tsx

```diff
@@ -440,33 +447,14 @@ If you provided a React element for the optionText prop, you must also provide t
     useEffect(() => {
         if (!multiple) {
             const optionLabel = getOptionLabel(selectedChoice);
-            if (typeof optionLabel === 'string') {
-                setFilterValue(optionLabel);
```

Comment:
I believe the issue happening because of this effect. 

Removing the call to `setFilterValue` solves the problem as long as `handleInputChange` (called by MUI's `onInputChange`) does it. 

I'm keeping the `useEffect` to throw an error if the user does not provide `inputText` along with `optionText`, but it could probably be moved in `handleInputChange` or `finalOnBlur`.